### DISCUSSION TITLE
How to recover Mirth with Root DB access - v4.0.0 Linux

### DISCUSSION THREAD

**Discussion by mochsner** [nextgenhealthcare/connect#5421] [Q&A] [answered]
I'm locked out of a mirth connect instance on linux, and trying to recover or reset a password. How the password was reset or broken, I have no idea - was working perfectly well last week, but now receive "Incorrect username or password"

Anyhow, we've tried permutations of things from the following posts, to no avail:
- [Old post about resetting a password using another / old DB ddesign, before person_password table](http://ibookmate.blogspot.com/2011/01/how-to-reset-mirth-admin-user.html)
- [Issue about resetting from the CLI](https://github.com/nextgenhealthcare/connect/issues/4879)
- [2011 forum post (linux instructions don't work - did salt or encryption SHA256 alg change?)](https://forums.mirthproject.io/forum/mirth-connect/support/3255-reset-admin-password)

At this point, I'm pretty much at a loss for how to reset this admin user (not admin/admin), or "reset" the server if needed. The mccommand all depends on having user credentials. Is there any (up-to-date) way to recover access to Mirth via DB?

**Comment by pacmano1** [comment]
```SQL
update person_password
set password = 'vZVIdOb0eKi6bNOfIwUr6CQ/5kFc1zwUksjTQtTu6W5EHk8xxON9Fg==\r+'
where person_id = 1;
```
Worked for me on two different instances.  One the latest docker container, another 4.0.1 on Ubuntu.   

Both use PostgreSQL for the back end.

**Reply by mochsner** [reply]
Simple (and slightly more straightforward than Jon's post which I followed), but for future readers, is this just supposed to reset the password to `admin`? If not, was there a repeatable process you followed to sha256 encrypt + base64 encode whatever pwd you set?

**Reply by ChristopherSchultz** [reply]
NextGen hasn't published the code to their password-mangler, so you just have to cross your fingers and hope it works. :)

**Comment by jonbartels** [comment, marked as answer]
You can see how the initial `admin/admin` password is created with this search - https://github.com/nextgenhealthcare/connect/search?q=INSERT+INTO+person_password 

You can find the relevant SQL file for your DB engine.

**Reply by pacmano1** [reply]
Much better suggestion @jonbartels

**Comment by jonbartels** [comment]
Regarding:

>  How the password was reset or broken, I have no idea - was working perfectly well last week

`SELECT * FROM events` and start adding WHERE clauses to find updates to the user settings or user password. IIRC the `person_password` also has a date column.

This may help you do some root-cause analysis.
